<?xml version="1.0"?>
<launch> 

  <node pkg="tf" type="static_transform_publisher" name="world_map_broadcaster_vis" args="0 0 0 0 0 0 world map 100"/>

  <group ns="visualization">
    
    <node pkg="rviz" name="rviz" type="rviz" args="-d $(find eurobot_sim)/config/visualization.rviz" output="screen"/>
    <param name="field_description" textfile="$(find eurobot_sim)/models/field.urdf" />
    
  </group>

  <group ns="main_robot">

    <!-- Robot model -->
    <param name="description" command="cat $(find eurobot_sim)/models/main_robot.urdf" />
    
    <!-- PARAMS -->
    
    <!-- robot -->
    <param name="robot_name" type="string" value="main_robot" />
    
    <!-- robot size -->
    <param name="dim_x" type="double" value="420" />
    <param name="dim_y" type="double" value="190" />
    <param name="dim_z" type="double" value="350" />

    <!-- initial coordinates -->
    <param name="start_x" type="double" value="293.0" />
    <param name="start_y" type="double" value="425.0" />
    <param name="start_a" type="double" value="4.71" />

  </group>

  <group ns="field">
  
    <node pkg="eurobot_sim" name="cubes_broadcaster" type="cubes_broadcaster.py"  output="screen"/>

    <!-- PARAMS -->

    <param name="color" type="string" value="orange" />

    <!-- cubes coordinates -->
    <param name="cube1c_x" type="double" value="850" />
    <param name="cube1c_y" type="double" value="540" />

    <param name="cube2c_x" type="double" value="300" />
    <param name="cube2c_y" type="double" value="1190" />

    <param name="cube3c_x" type="double" value="1100" />
    <param name="cube3c_y" type="double" value="1500" />

    <param name="cube4c_x" type="double" value="1900" />
    <param name="cube4c_y" type="double" value="1500" />

    <param name="cube5c_x" type="double" value="2700" />
    <param name="cube5c_y" type="double" value="1190" />

    <param name="cube6c_x" type="double" value="2150" />
    <param name="cube6c_y" type="double" value="540" />

  </group>

</launch>
